src/runtime/string.go

```diff
@@ -89,6 +89,21 @@ func stringToBytes(x _string) (slice struct {
 	return
 }
 
+// Convert a string to []rune slice.
+func stringToRunes(s string) []rune {
+	var n = 0
+	for range s {
+		n++
+	}
+	var r = make([]rune, n)
+	n = 0
+	for _, e := range s {
```

Comment:
Seems like you could replace the `range` here with:

```
for i := 0; i < n; i++ {
	r[i] = s[i]
}
```

Reply:
Unfortunately that doesn't work, indexing into a string returns a byte, not a rune.